File: examples/docs-code-snippets/src/main/java/examples/java/application/InquiryExample.java
package examples.java.application;

import com.ing.baker.runtime.javadsl.Baker;

public class InquiryExample {

    private final Baker baker;

    public InquiryExample(Baker baker) {
        this.baker = baker;
    }

    public void example(String recipeInstanceId) {
        var ingredient = baker.getIngredient(recipeInstanceId, "orderId").join();

        var ingredients = baker.getIngredients(recipeInstanceId).join();

        var events = baker.getEvents(recipeInstanceId).join();

        var eventNames = baker.getEventNames(recipeInstanceId).join();

        var recipeInstanceState = baker.getRecipeInstanceState(recipeInstanceId).join();
    }
}

File: examples/docs-code-snippets/src/main/kotlin/examples/kotlin/application/InquiryExample.kt
package examples.kotlin.application

import com.ing.baker.runtime.kotlindsl.Baker

class InquiryExample(private val baker: Baker) {

    suspend fun example(recipeInstanceId: String) {
        val ingredient = baker.getIngredient(recipeInstanceId, "orderId")

        val ingredients = baker.getIngredients(recipeInstanceId)

        val events = baker.getEvents(recipeInstanceId)

        val eventNames = baker.getEventNames(recipeInstanceId)

        val recipeInstanceState = baker.getRecipeInstanceState(recipeInstanceId)
    }
}